src/lib/y2users/config_manager.rb

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "singleton"
+require "y2users/config"
+
+module Y2Users
+  # Holds references to different configuration instances.
+  # It is not mandatory for config to be registered here.
+  class ConfigManager
+    include Singleton
+
+    def initialize
+      @register = {}
+    end
+
+    # Register config as given id
+    # @param [Config] config
+    # @param [Symbol] as id of config
+    # @note if given id is already registered, it is overwritten
+    # rubocop:disable Naming/UncommunicativeMethodParamName
```

Comment:
NP: What about configuring rubocop to accept two characters length parameters? By default, `MinNameLength` is 3, see https://www.rubydoc.info/gems/rubocop/0.53.0/RuboCop/Cop/Naming/UncommunicativeMethodParamName.

Reply:
I am not sure. Usually I prefer to not deviate unless there are good reason. It is also possible to add "as" as recognized names, I am just not sure if it is generally good name beside this use case.